version: "3"

services:    
    
    feedo-test:
      profiles: [test]
      container_name: feedfarer-test
      depends_on: [redis]
      restart: "no"
      build: 
        context: .
        dockerfile: Dockerfile-test
      environment:
        - API_KEY
        - TELEGRAM_TOKEN
        - BASE_URL
        - ALERT_CHATID
        - WEBHOOK_URL
        - MONGO_CONN_STRING
    
    feedo:
      profiles: [prod]
      image: ghcr.io/why-not-try-calmer/feedo:latest
      depends_on: [redis]
      restart: on-failure
      build: .
      command: ./feefarer-exe
      environment:
        API_KEY: ${API_KEY}
        TELEGRAM_TOKEN: ${TELEGRAM_TOKEN}
        BASE_URL: ${BASE_URL}
        ALERT_CHATID: ${ALERT_CHATID}
        WEBHOOK_URL: ${WEBHOOK_URL}
        MONGO_CONN_STRING: ${MONGO_CONN_STRING}
        PORT: 8000
    
    mongo:
      image: mongo:5.0 # `auth` doesn't work with 6.0 and beyond!
      restart: on-failure
      environment:
        - MONGO_INITDB_ROOT_USERNAME
        - MONGO_INITDB_ROOT_PASSWORD
      volumes:
       - mongo-data:/data/db

    nginx:
      profiles: [prod]
      image: nginx:latest
      restart: on-failure
      volumes:
        - ./nginx.conf:/etc/nginx/nginx.conf:ro
        - ./private.key:/etc/nginx/private.key:ro
        - ./cert.pem:/etc/nginx/cert.pem:ro
      ports:
        - "80:80"
        - "443:443"
      depends_on:
      - redis
      - feedo
  
    redis:
      image: redis:latest
      expose: [6379]
      restart: on-failure
      command: redis-server --maxmemory 250mb --maxmemory-policy volatile-lfu
      volumes:
        - redis-data:/data/redis-store

volumes:
  mongdb-data:
  redis-data: